From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Hilman <khilman@baylibre.com>
Date: Thu, 4 Feb 2021 11:57:59 -0800
Subject: ANDROID: GKI: enable networking; add hidden network configs

Drop the explicit disable of CONFIG_ETHERNET from gki_defconfig so
wired networking support is enabled by default.

Also, when testing GKI with out-of-tree wired networking modules, it
was discovered that a few hidden Kconfigs are required.  Add them here
so out of tree net drivers can be built in a separate tree.

Bug: 179406580
Change-Id: Ia66742e514ae1a3c9f8d0ec1d92eb40b68d12ce7
Signed-off-by: Kevin Hilman <khilman@baylibre.com>
[joneslee: Removed defconfig elements]
Signed-off-by: Lee Jones <lee.jones@linaro.org>
---
 init/Kconfig.gki | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -185,6 +185,15 @@ config GKI_HIDDEN_HYPERVISOR_CONFIGS
 	  config, required by various SoC platforms. This is usually
 	  selected by XEN or S390.
 
+config GKI_HIDDEN_NET_CONFIGS
+	bool "Hidden networking configuration needed for GKI"
+	select PAGE_POOL
+	select NET_PTP_CLASSIFY
+	help
+	  Dummy config option used to enable the networking hidden
+	  config, required by various SoC platforms.
+
+
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
 # b) building drivers that are not modularizable
@@ -211,6 +220,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_GPU_CONFIGS
 	select GKI_HIDDEN_IRQ_CONFIGS
 	select GKI_HIDDEN_HYPERVISOR_CONFIGS
+	select GKI_HIDDEN_NET_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
